src/components/Table/TableCell.tsx

```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
     cell: any;
-    state: TableState<any>;
+    selectionMode: string;
     type?: TableCellType;
+    isChecked?: boolean;
+    selectedRows: Key[];
+    setSelectedRows?: (ids?: Key[]) => void;
 };
 
-export const TableCell: FC<TableCellProps> = ({ cell, state, type = TableCellType.Default }) => {
+export const TableCell: FC<TableCellProps> = ({
+    cell,
+    selectionMode,
+    type = TableCellType.Default,
+    isChecked = false,
+    selectedRows,
+    setSelectedRows,
+}) => {
     const ref = useRef<HTMLTableCellElement | null>(null);
-    const { gridCellProps } = useTableCell({ node: cell }, state, ref);
-    const { checkboxProps } = useTableSelectionCheckbox({ key: cell.parentKey }, state);
-    const inputRef = useRef(null);
-    const {
-        inputProps: { checked },
-    } = useCheckbox(checkboxProps, useToggleState(checkboxProps), inputRef);
-    const { isFocusVisible, focusProps } = useFocusRing();
 
     if (type === TableCellType.Checkbox) {
         const { key } = cell;
+        const handleChange = () => {
+            if (!setSelectedRows) {
+                return;
+            }
+
+            if (isChecked) {
+                const filteredRows = selectedRows.filter((row) => row !== cell.parentKey);
+                setSelectedRows(filteredRows);
+                return;
+            }
+            if (selectionMode === SelectionMode.SingleSelect) {
+                setSelectedRows([cell.parentKey]);
+            } else {
+                setSelectedRows([...selectedRows, cell.parentKey]);
+            }
+        };
 
         return (
             <td
-                {...gridCellProps}
+                role="cell"
                 ref={ref}
                 className={merge([
                     'tw-pl-8 tw-py-4 tw-pr-4 tw-border-l-4',
-                    checked ? 'tw-border-violet-60' : 'tw-border-transparent',
+                    isChecked ? 'tw-border-violet-60' : 'tw-border-transparent',
                 ])}
                 data-test-id="table-select-cell"
             >
-                <Checkbox
+                <CheckboxCmp
                     value={key}
                     ariaLabel={cell['aria-label'] || key}
-                    state={checked ? CheckboxState.Checked : CheckboxState.Unchecked}
+                    state={isChecked ? CheckboxState.Checked : CheckboxState.Unchecked}
+                    onChange={handleChange}
                 />
             </td>
         );
     }
 
     return (
         <td
-            {...mergeProps(gridCellProps, focusProps)}
+            role="cell"
             ref={ref}
             className={merge([
                 'tw-p-4 tw-font-normal tw-text-xs focus:tw-outline-none',
-                checked ? 'tw-text-black-100 dark:tw-text-white' : 'tw-text-black-80 dark:tw-text-black-20',
-                isFocusVisible && FOCUS_STYLE_INSET,
+                isChecked ? 'tw-text-black-100 dark:tw-text-white' : 'tw-text-black-80 dark:tw-text-black-20',
```

Comment:
the classes here should be changed to use `Alias tokens` instead and avoid to use `dark` prefix.
We shouldn't have things like `tw-text-black-100` for example or violet-...

Reply:
Will be resolved in another PR as per discussion in slack